models/stripe__invoice_line_items.sql

```diff
@@ -64,7 +64,10 @@ select
     customer.customer_id
 
     {% if var('using_subscriptions', True) %}
-    ,subscription.subscription_id,
+    ,invoice_line_item.proration, 
+    extract(epoch from (invoice_line_item.period_end::timestamp - invoice_line_item.period_start::timestamp + (plan.interval_count || ' MONTH')::INTERVAL)) as estimated_service_period_start,
```

Comment:
shouldn't we consider `plan_interval` here?

```suggestion
    extract(epoch from (invoice_line_item.period_end::timestamp - invoice_line_item.period_start::timestamp + (plan.interval_count || ' ' || plan.plan_interval)::INTERVAL)) as estimated_service_period_start,
```

Reply:
`plan_interval` is a string unless we transform it to integer

Reply:
My suggestion is change

```
 (plan.interval_count || ' MONTH')::INTERVAL
```

to

```
 (plan.interval_count || ' ' || plan.plan_interval)::INTERVAL
```

Which will be evaulated as '1 month', or '1 year', etc.

Reply:
Also, I believe `estimated_full_service_period` is wrong. I think it should be something like this:

```sql
case 
  when invoice_line_item.proration then 
     extract(epoch from (invoice_line_item.period_end - invoice_line_item.period_start)) as estimated_full_service_period
  else
     --full duration of subscription
end
```

Reply:
Actually, it's the inverse. When an item is a proration, we need to calculate the full period by looking at `period_end - plan_interval`, and when it's not a proration, do what you said above. But you need to revert the order and put period_start before period_end